File: iosApp/iosApp/Analytics/AnalyticsService.swift
//
//  AnalyticsService.swift
//  DevFest Nantes
//
//  Created by Stéphane Rihet on 28/09/2022.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation
import SwiftUI

protocol AnalyticsService {

    // MARK: - Events
    
    func eventAddToFavorite(from page: AnalyticsPage, sessionId: String, fav: Bool)
    
    func eventLinkCodeOfConductOpened()
    
    func eventLinkDevFestWebsiteOpened()
    
    func eventLinkFacebookOpened()
    
    func eventLinkLinkedinOpened()
    
    func eventLinkLocalCommunitiesOpened()
    
    func eventLinkPartnerOpened(partnerURL: String)
    
    func eventLinkTwitterOpened()
    
    func eventLinkYoutubeOpened()
    
    func eventNavigationClicked()

    func eventSessionOpened(sessionId: String)

    func eventSpeakerSocialLinkOpened(speaker: String, url: String)


    // MARK: - Pages

    func pageEvent(page: AnalyticsPage)

}

File: shared/src/commonMain/kotlin/com/gdgnantes/devfest/analytics/AnalyticsService.kt
package com.gdgnantes.devfest.analytics

import com.gdgnantes.devfest.model.SocialType

@Suppress("TooManyFunctions")
interface AnalyticsService {
    fun eventBookmark(page: AnalyticsPage, sessionId: String, bookmarked: Boolean)

    fun eventFilter()

    fun eventLinkCodeOfConductOpened()

    fun eventLinkDevFestWebsiteOpened()

    fun eventLinkFacebookOpened()

    fun eventLinkGithubOpened()

    fun eventLinkLinkedinOpened()

    fun eventLinkLocalCommunitiesOpened()

    fun eventLinkPartnerOpened(partnerName: String)

    fun eventLinkSupportOpened()

    fun eventLinkTwitterOpened()

    fun eventLinkYoutubeOpened()

    fun eventFeedbackClicked(openFeedbackId: String)

    fun eventSessionOpened(sessionId: String)

    fun eventSpeakerOpened(speakerId: String)

    fun eventSpeakerSocialLinkOpened(speakerId: String, type: SocialType)

    fun eventVenueNavigationClicked()

    fun eventVenueFloorPlanClicked()

    fun pageEvent(page: AnalyticsPage, className: String? = null)
}
